containers/ome-tiff-offsets/context/main.py

```diff
@@ -2,26 +2,35 @@
 from glob import glob
 from pathlib import Path
 from os import makedirs
+from os.path import splitext
 import json
+from os.path import dirname
+from os.path import exists
+from os import makedirs
 
 from skimage.external import tifffile
 
-
 def get_offsets(tiff_filepath):
     with tifffile.TiffFile(tiff_filepath) as tif:
         offsets = [page._offset for page in tif.pages]
     return offsets
 
 def main(input_dir, output_dir):
     makedirs(output_dir, exist_ok=True)
-    for input_path in glob(input_dir + '/*.ome.tiff') + glob(input_dir + '/*.ome.tif'):
+
+    #find all OME.TIFFs in the input directory
+    for input_path in glob(input_dir + '/**/*ome.tif', recursive=True) + glob(input_dir + '/**/*ome.tiff', recursive=True):
         offsets = get_offsets(input_path)
-        # Truncate `.ome.tiff`.
-        input_name = Path(input_path).with_suffix('').with_suffix('').name
-        output_path = Path(output_dir) / str(input_name + '.offsets.json')
+
+        #creates output path for each OME.TIFF
+        output_path = dirname( str(Path(output_dir) / Path( input_path ).relative_to( input_dir )) )
+        if not exists(output_path):
+            makedirs(output_path)
+
+        #sets output filename for JSON file and dumps to disk
+        output_path = str( Path(output_dir) / Path( input_path ).relative_to( input_dir ).with_suffix('').with_suffix('') ) + '.offsets.json'
```

Comment:
Not clear what `with_suffix('')` does.

Reply:
I was recycling the original code. Return a new path with the suffix changed, in this case it removes `ome.tif`.